collect_app/src/main/java/org/odk/collect/android/formentry/audit/AuditConfig.java

```diff
@@ -52,13 +52,15 @@
     private final boolean isTrackingChangesEnabled;
 
     private final boolean isIdentifyUserEnabled;
+    private final boolean isTrackChangesReasonEnabled;
 
-    public AuditConfig(String mode, String locationMinInterval, String locationMaxAge, boolean isTrackingChangesEnabled, boolean isIdentifyUserEnabled) {
+    public AuditConfig(String mode, String locationMinInterval, String locationMaxAge, boolean isTrackingChangesEnabled, boolean isIdentifyUserEnabled, boolean isTrackChangesReasonEnabled) {
```

Comment:
We have more and more parameters maybe it would be good to use the builder pattern?

Reply:
Yeah good shout. I'll generate one.

Reply:
Kotlin would solve it right? But we use Java.

Reply:
Haha yes it would! The [`data` classes](https://kotlinlang.org/docs/reference/data-classes.html) automatically get a `copy` function that works just like a builder because of the named/optional parameters.